src/components/ListItem.jsx

```diff
@@ -11,22 +11,28 @@ export function ListItem({
 	itemId,
 	name,
 	isChecked,
+	dateCreated,
 	dateLastPurchased,
+	dateNextPurchased,
+	totalPurchases,
 }) {
 	const DAYINMS = 86400000;
 	// sync up checked or not checked data from the database to the page upon page refresh
 	const [isPurchased, setIsPurchased] = useState(isChecked);
 
 	useEffect(() => {
-		// only update item when the checkbox has been checked. Don't update when the page just loaded and "isPurchased" is changed accordingly
 		if (isChecked !== isPurchased) {
```

Comment:
Here's where you could add a bit more logic to prevent the issue with incrementing too many times if a user is click happy (see gif [here](https://cdn.zappy.app/8197aa3cbf65e9f08b7c81c8aad2b4b7.gif). In addition to checking if isChecked !== isPurchased, you could confirm the timestamp of the last purchase date with now() and only increment if the timestamps are say..50k milliseconds apart. This might better ensure that the user is unable to add multiple accidentally, but if they have multiple trips in a day for whatever reason, they're not restricted from recording true additional purchases.

Reply:
Yes! That's a good idea! With this week's issue done, I think how to solve this "click-happy-user" problem becomes more and more clear!

Reply:
@chelseychubb Thanks for the feedback! We have added this before merging with main just now by updating the conditional. Now the app takes into account the last time item was checked off and does not increment in that case, while keeping the `isChecked` value synced between the app and Firebase.